File: java/Facade/complete/Facade.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

//class for getting the seconds portion of the current time
class Seconds{
  public String get(){
    DateTimeFormatter formatter = DateTimeFormatter.ISO_DATE_TIME;
    LocalDateTime my_time = LocalDateTime.now();
    return my_time.format(formatter).substring(17, 19);
  }
}

//class for getting the minutes portion of the current time
class Minutes{
  public String get(){
    DateTimeFormatter formatter = DateTimeFormatter.ISO_DATE_TIME;
    LocalDateTime my_time = LocalDateTime.now();
    return my_time.format(formatter).substring(14, 16);
  }
}

//class for getting the hours portion of the current time
class Hours{
  public String get(){
    DateTimeFormatter formatter = DateTimeFormatter.ISO_DATE_TIME;
    LocalDateTime my_time = LocalDateTime.now();
    return my_time.format(formatter).substring(11, 13);
  }
}

/*
a simple interface for getting the current time, combines Seconds,
Hours and Minutes classes
*/
public class Facade{

  public String get_time(){
    Seconds sec = new Seconds();
    Minutes min = new Minutes();
    Hours hrs = new Hours();

    return sec.get() + ":" + min.get() + ":" + hrs.get();
  }

}

File: python/Facade/complete/Facade.py
from datetime import datetime

class Seconds(object):
    '''interface for getting the seconds part of the current time'''
    def get(self):
        return str(datetime.now().time())[6:8]

class Minutes(object):
    '''interface for getting the minutes portion of the current time'''
    def get(self):
        return str(datetime.now().time())[3:5]

class Hours(object):
    '''interface for getting the hours portion of the current time'''
    def get(self):
        return str(datetime.now().time())[0:2]

class Facade(object):
    '''
    a simple interface for getting the current time, combines Seconds,
    Hours and Minutes classes
    '''

    def get_time(self):
        sec = Seconds().get()
        min = Minutes().get()
        hrs = Hours().get()

        return hrs + ":" + min + ":" + sec
